frontend/src/charts/TableChart.tsx

```diff
@@ -1,70 +1,135 @@
 import React from "react";
```

Comment:
You could also import useMemo here so you don't have to do React.useMemo below:
import React, {useMemo}

just a style preference. up to you